packages/gitbook/src/components/DocumentView/Expandable/Details.tsx

```diff
@@ -0,0 +1,98 @@
+'use client';
+
+import { useParams } from 'next/navigation';
+import React from 'react';
+
+import { ClassValue, tcls } from '@/lib/tailwind';
+
+function useHash() {
+    const params = useParams();
+    const [hash, setHash] = React.useState<string>(global.location?.hash?.slice(1));
+    React.useEffect(() => {
+        function updateHash() {
+            setHash(global.location?.hash?.slice(1));
+        }
+        global.addEventListener('hashchange', updateHash);
+        updateHash();
+        return () => global.removeEventListener('hashchange', updateHash);
+    }, [params]);
+    return hash;
+}
+
+/**
+ * Details component rendered on client so it can expand dependent on url hash changes.
+ */
+export function Details(props: {
+    children: React.ReactNode;
+    id: string;
+    contentIds?: string[];
+    open?: boolean;
+    className?: ClassValue;
+}) {
+    const { children, id, open, className } = props;
+
+    const detailsRef = React.useRef<HTMLDetailsElement>(null);
+
+    const [anchorElement, setAnchorElement] = React.useState<Element | null | undefined>();
+
+    const hash = useHash();
+    React.useEffect(() => {
+        if (!hash) {
+            return;
+        }
+        const descendant =
+            hash === id ? detailsRef.current : detailsRef.current?.querySelector(`#${hash}`);
```

Comment:
```suggestion
            hash === id ? detailsRef.current : detailsRef.current?.getElementById(hash);
```

I had issues in the past with using hash in query selectors as it needs to be properly encoded.

Reply:
`getElementById` is only on the document but can combine it with contains for the context of the element.